zulip_bots/zulip_bots/test_lib.py

```diff
@@ -172,3 +172,14 @@ def mock_http_conversation(self, test_name):
     def mock_config_info(self, config_info):
         # type: (Dict[str, str]) -> Any
         return patch('zulip_bots.test_lib.StubBotHandler.get_config_info', return_value=config_info)
+
+    def mock_request_message(self, mock_message):
+        # type: (Dict[str, object]) -> Any
+        message = dict(
+            display_recipient=mock_message.get('display_recipient', 'foo_stream'),
+            sender_email=mock_message.get('sender_email', 'foo@example.com'),
+            sender_full_name=mock_message.get('sender_full_name', 'Foo Test User'),
+            sender_id=mock_message.get('sender_id', '123'),
+            content=mock_message.get('content', ''),
+        )
+        return patch('zulip_bots.test_lib.BotTestCase.make_request_message', return_value=message)
```

Comment:
This looks like a big function for something that could be accomplished with one additional line of code in the real test. I recommend to proceed like stated in the description of `make_request_message`:
> This is mostly used internally but
tests can override this behavior by
mocking/subclassing.

So in short, just directly patch the function in the test where you are using it.